fabric-fluid-v1/src/main/java/net/fabricmc/fabric/api/fluids/v1/world/SidedFluidContainer.java

```diff
@@ -0,0 +1,62 @@
+package net.fabricmc.fabric.api.fluids.v1.world;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.FluidDrainable;
+import net.minecraft.block.FluidFillable;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.fluid.Fluids;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.BlockView;
+import net.minecraft.world.World;
+import net.minecraft.world.WorldAccess;
+
+import net.fabricmc.fabric.Action;
+import net.fabricmc.fabric.api.fluids.v1.container.FluidContainer;
+import net.fabricmc.fabric.api.fluids.v1.container.volume.FluidVolume;
+import net.fabricmc.fabric.api.fluids.v1.container.volume.SimpleFluidVolume;
+import net.fabricmc.fabric.api.fluids.v1.math.Drops;
+
+/**
+ * A container with different fluid containers per face, can be implemented on entities, and blocks (does not check block entities!).
+ */
+public interface SidedFluidContainer extends FluidFillable, FluidDrainable {
+	@Override
+	default Fluid tryDrainFluid(WorldAccess world, BlockPos pos, BlockState state) {
+		if(world instanceof World) {
+			FluidVolume volume = this.getContainer((World) world, pos, null).drain(Drops.getBucket(), Action.SIMULATE);
+			if(volume.getFluid() != Fluids.EMPTY && volume.getAmount() == Drops.getBucket()) {
+				return this.getContainer((World) world, pos, null).drain(Drops.getBucket(), Action.PERFORM).getFluid();
+			}
+		}
+		return Fluids.EMPTY;
+	}
+
+	/**
+	 * Get the container for the given face.
```

Comment:
You've named a `Direction` as both `face` and `direction` so far, I suggest picking one and going with it.